src/Components/getList.js

```diff
@@ -27,16 +27,67 @@ function EmptyList() {
   );
 }
 
-function FullList(props) {
+
+const FullList = (props)=> {
   const [isChecked, setCheck] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const handleChange = e => {
+    setSearchTerm(e.target.value);
+  };
+
+  const clearSearch = () => {
+    setSearchTerm("");
+  };
+
+  const searchList = props.data.filter(item => {
+    return item.itemName.toLowerCase().includes(searchTerm.toLowerCase());
+  });
+
 
 
   return (
     <div className="grocery-list">
       <h1>Groceries</h1>
-      <ul className="tl f2">
-        {props.data.map(item => (
+      <ul className="tl f2 check-list">
+      <li className="bg-white pa1 shadow">
+      <div className="filter-input">
+      <input 
+         type="text"
+         placeholder="Filter List"
+         value={searchTerm}
+         onChange={handleChange}
+       />
+       <button className="filter-button" onClick={clearSearch}> X </button>
+       </div>
+       </li>
+
+    
+        {searchTerm
+          ? searchList.map(item => (
+            <li className="bg-white pa1 shadow" key={item.id}>
+               <div className="checkbox-input">
+            <label>
+                    <input 
+                      type="checkbox"
+                        value= {isChecked}
+                        onChange={() => setCheck(checked => !checked)}
+                      defaultChecked={
+                        item.lastPurchasedDate
+                          ? Date.now() / 1000 - item.lastPurchasedDate.seconds <
+                            86400
```

Comment:
I'd like us to remove the **magic numbers** moving forward, and instead store them in variables with meaningful names.

From this, I know what's 86400 (seconds in a day) but I don't know what's `1000`.

```
const SECONDS_IN_A_DAY = 86400;
const SOMETHING = 1000;
```